src/app/features/switch-account-drawer/switch-account-drawer.tsx

```diff
@@ -37,19 +37,30 @@ export const SwitchAccountDrawer = memo(() => {
     return <AccountListUnavailable />;
   }
 
-  return isShowing ? (
-    <ControlledDrawer title="Select account" isShowing={isShowing} onClose={onClose}>
-      <Box mb={whenWallet({ ledger: 'space.04', software: '' })}>
-        <SwitchAccountList
-          currentAccountIndex={currentAccountIndex}
-          handleClose={onClose}
-          addressesNum={stacksAddressesNum || btcAddressesNum}
-        />
-        {whenWallet({
-          software: <CreateAccountAction onCreateAccount={onCreateAccount} />,
-          ledger: <></>,
-        })}
-      </Box>
-    </ControlledDrawer>
-  ) : null;
+  // #4370 SMELL without this early return the wallet crashes on new install with:
+  // Wallet is neither of type `ledger` nor `software`
+  // FIXME - this shouldn't be rendered until necessary
+  if (!isShowing) return null;
+
+  return (
+    <Modal
+      title={<styled.h1 textStyle="heading.05">Select account</styled.h1>} // #4370 FIXME: Headers passing JSX as a hack for Select Account that needs to be fixed
```

Comment:
Are the FIXME comments going to be left or addressed in this PR still?

Reply:
I'll be fixing them pretty soon, depending on priorities but this week. I thought it good to get eyes on work so far at least. 

For the next part of this I need to refactor all the headers so that's the next batch of work